### PR TITLE
Allows you to use the ability of vatbeasts when riding them

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 add: Allows you to use the ability of vatbeasts when riding them
/:cl:

because riding is kinda weak.

### CODE REVIEW

File: code/datums/components/riding.dm [hunk 1]
```diff
@@ -50,6 +54,32 @@
...
+
+///Gives the rider the riding parent's abilities
+/datum/component/riding/proc/setup_abilities(mob/living/M)
+
+	if(!istype(parent, /mob/living))
+		return
+
+	var/mob/living/ridden_creature = parent
+
+	for(var/i in ridden_creature.abilities)
```
Comment (Jared-Fogle) [inline code/datums/components/riding.dm:68]: Better variable names please.

File: code/modules/spells/spell.dm [hunk 2]
```diff
@@ -14,9 +14,11 @@
 	var/action_icon_state = "spell_default"
 	var/action_background_icon_state = "bg_spell"
 	var/base_action = /datum/action/spell_action
+	var/mob/living/owner
 
-/obj/effect/proc_holder/Initialize()
+/obj/effect/proc_holder/Initialize(mob/living/owner)
 	. = ..()
+	owner = owner
```
Comment (Jared-Fogle) [inline code/modules/spells/spell.dm:21]: There is no way this works.

File: code/datums/components/riding.dm (hunk 1 above)
Comment (Qustinnus) [reply code/datums/components/riding.dm:68]: check the line below, this is normal behavior for these kinds of loops.

File: code/modules/spells/spell.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/spells/spell.dm]: yeah i added a src.owner to the first one my bad

**Comment by ATH1909** [comment]
Yay, vatbeasts are actually useful now!

Would it be possible to also add a button to invoke the (default) ranged attacks of a mob you're riding? Like, if you're somehow riding a hivebot or a magicarp, you'd be able to push a button to switch to "projectile firing mode" or something.

**Comment by Qustinnus** [comment]
probably possible but mob code is ass + im lazy

File: code/datums/components/riding.dm [hunk 3]
```diff
@@ -50,6 +54,32 @@
...
+		var/obj/effect/proc_holder/proc_holder = i
+		M.AddAbility(proc_holder)
```
Comment (ATH1909) [inline code/datums/components/riding.dm:70]: by the way, spells don't count as abilities, right? if I somehow get a wizard into the body of a vatbeast, I can't copy his spells, right?

Comment (Qustinnus) [reply code/datums/components/riding.dm:70]: dunno dont think so

File: code/datums/components/riding.dm (hunk 1 above)
Comment (Rohesie) [reply code/datums/components/riding.dm:68]: Yep, the line bellow has an actually readable variable name.

**Pull request merged**